megamek/src/megamek/client/bot/princess/Princess.java

```diff
@@ -928,6 +1032,318 @@ protected Vector<Minefield> calculateMinefieldDeployment() {
+                                              FiringPlan planOfAttack,
+                                              int maximumToHit,
+                                              boolean rearAttack,
+                                              boolean includeHead) {
+        int aimLocation = Mech.LOC_NONE;
+
+        // If none of the weapons being fired can be aimed, don't bother finding the best location
+        List<WeaponFireInfo> workingShots = planOfAttack.
+                stream().
+                filter(curShot ->
+                Compute.allowAimedShotWith(curShot.getWeapon(),
+                        target.isImmobile() ? AimingMode.IMMOBILE : AimingMode.TARGETING_COMPUTER)).
+                collect(Collectors.toList());
+
+        if (workingShots.isEmpty()) {
+            return aimLocation;
+        }
+
+        int lowestArmor = 1000;
+        List<Integer> rankedLocations = new ArrayList<>();
+
+        // Aiming for the head can only be done against an immobile Mech, and takes a penalty.
+        // Don't aim for the head for anti-Mech attacks except after swarming.
+        if (includeHead &&
+                target.isImmobile() &&
+                planOfAttack.get(0).getWeapon().getShortName() != Infantry.LEG_ATTACK &&
+                planOfAttack.get(0).getWeapon().getShortName() != Infantry.SWARM_MEK &&
+                planOfAttack.get(0).getWeapon().getShortName() != Infantry.STOP_SWARM) {
+            aimLocation = Mech.LOC_HEAD;
+            if (planOfAttack.stream().anyMatch(curFire -> curFire.getToHit().getValue() + 7 > maximumToHit)) {
+                aimLocation = Mech.LOC_NONE;
+            } else {
+                return aimLocation;
+            }
+        }
+
+        // Limit leg attack aimed shots to the legs
+        if (workingShots.get(0).getWeapon().getShortName() != Infantry.LEG_ATTACK) {
+
+            // Consider arm locations if they have a 'big' weapon
+            for (WeaponMounted curWeapon : target.getWeaponList().
+                    stream().
+                    filter(w -> w.isOperable() &&
+                    w.getType().getDamage(5) >= 10).collect(Collectors.toSet())) {
+
+                if (!rankedLocations.contains(Mech.LOC_RARM) &&
+                        curWeapon.getLocation() == Mech.LOC_RARM &&
+                        target.getInternal(Mech.LOC_RARM) > 0) {
+                    rankedLocations.add(Mech.LOC_RARM);
+                } else if (!rankedLocations.contains(Mech.LOC_LARM) &&
+                        curWeapon.getLocation() == Mech.LOC_LARM &&
+                        target.getInternal(Mech.LOC_LARM) > 0) {
+                    rankedLocations.add(Mech.LOC_LARM);
+                }
+                if (rankedLocations.contains(Mech.LOC_RARM) && rankedLocations.contains(Mech.LOC_LARM)) {
+                    break;
+                }
+
+            }
+
+            // Favor right torso over left, except if right torso is completely gone
```

Comment:
Why?

Reply:
Most Mechs are 'right handed' i.e. if they have a big gun it's in the right arm or right torso.  By favoring the right torso, it's more likely that at least one main weapon will be damaged if not outright removed by the arm falling off.  The side torsos are usually lower armor than the CT, so they get checked first.

I'll see about adding something descriptive to cover that reasoning.